### PR TITLE
move interface to adapters

### PR DESCRIPTION
**Pull request by marioreggiori**

### CODE REVIEW

File: src/storage/adapters/gridfs.ts [hunk 1]
```diff
@@ -1,17 +1,57 @@
...
+            });
+        })
     }
-    async get(id: string): Promise<StorageFile> {
-        throw new Error("Method not implemented.");
-    }
-    async list(id: string): Promise<StorageFile[]> {
-        throw new Error("Method not implemented.");
+
+    async get(id: string, stream: WriteStream): Promise<void> {
```
Comment (flexwie) [inline src/storage/adapters/gridfs.ts:41]: Wouldnt it be better if we returned the stream itself (or a promised version of it)?

File: src/storage/adapters/interface.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
+import { ReadStream, WriteStream } from 'fs';
+import { Writable } from 'stream';
 
-interface StorageAdapter {
-    add(id: string): Promise<string>;
-    get(id: string): Promise<StorageFile>;
+export default interface StorageAdapter {
+    add(id: ReadStream): Promise<string>;
```
Comment (flexwie) [inline src/storage/adapters/interface.ts]:
```suggestion
    add(id: string, stream: ReadStream): Promise<string>;
```

File: src/storage/adapters/s3bucket.ts [hunk 3]
```diff
@@ -9,17 +12,14 @@ class S3Bucket implements StorageAdapter {
             sslEnabled: true,
         });
     }
-    async add(id: string): Promise<string> {
+    async add(id: ReadStream): Promise<string> {
```
Comment (flexwie) [inline src/storage/adapters/s3bucket.ts]:
```suggestion
    async add(id: string, stream: ReadStream): Promise<string> {
```

**Review by flexwie — CHANGES_REQUESTED** [review]

File: src/storage/adapters/gridfs.ts (hunk 1 above)
Comment (marioreggiori) [reply src/storage/adapters/gridfs.ts:41]: Couldn't get it to work.

**Review by flexwie — APPROVED** [review]

**Pull request merged**